homeassistant/components/elkm1/__init__.py

```diff
@@ -65,32 +67,34 @@ def _elk_value(val):
     return (start, end)
 
 
-CONFIG_SCHEMA_SUBDOMAIN = vol.Schema({
+DEVICE_SCHEMA_SUBDOMAIN = vol.Schema({
     vol.Optional(CONF_ENABLED, default=True): cv.boolean,
     vol.Optional(CONF_INCLUDE, default=[]): [_elk_range_validator],
     vol.Optional(CONF_EXCLUDE, default=[]): [_elk_range_validator],
 })
 
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PREFIX, default=''): cv.string,
+    vol.Optional(CONF_USERNAME, default=''): cv.string,
+    vol.Optional(CONF_PASSWORD, default=''): cv.string,
+    vol.Optional(CONF_TEMPERATURE_UNIT, default='F'):
+    cv.temperature_unit,
+    vol.Optional(CONF_AREA, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_COUNTER, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_KEYPAD, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_OUTPUT, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_PLC, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_SETTING, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_TASK, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_THERMOSTAT, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+    vol.Optional(CONF_ZONE, default={}): DEVICE_SCHEMA_SUBDOMAIN,
+}, _host_validator)
+
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema(
-        {
-            vol.Required(CONF_HOST): cv.string,
-            vol.Optional(CONF_USERNAME, default=''): cv.string,
-            vol.Optional(CONF_PASSWORD, default=''): cv.string,
-            vol.Optional(CONF_TEMPERATURE_UNIT, default='F'):
-                cv.temperature_unit,
-            vol.Optional(CONF_AREA, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_COUNTER, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_KEYPAD, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_OUTPUT, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_PLC, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_SETTING, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_TASK, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_THERMOSTAT, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-            vol.Optional(CONF_ZONE, default={}): CONFIG_SCHEMA_SUBDOMAIN,
-        },
-        _host_validator,
-    )
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [DEVICE_SCHEMA])
```

Comment:
I would recommend removing the key `CONF_DEVICES` to remove the breaking change (similar to how the [ZoneMinder component is implemented](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/zoneminder/__init__.py)).

This would then look like:

```
CONFIG_SCHEMA = vol.Schema({
    DOMAIN: vol.All(cv.ensure_list, [DEVICE_SCHEMA])
}, extra=vol.ALLOW_EXTRA)
```